packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -0,0 +1,535 @@
+    * @param _token Address of the token to be queried
+    * @return True if the juror has already claimed the corresponding token fees
+    */
+    function hasJurorClaimed(uint256 _periodId, address _juror, address _token) external view returns (bool) {
+        Period storage period = periods[_periodId];
+        return _hasClaimedJurorFees(period, _juror, _token);
+    }
+
+    /**
+    * @dev Check if a given juror has already claimed the owed fees for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _juror Address of the juror being queried
+    * @param _tokens List of token addresses to be queried
+    * @return List of status to tell whether the corresponding token was claimed by the juror
+    */
+    function hasJurorClaimedMany(uint256 _periodId, address _juror, address[] calldata _tokens) external view returns (bool[] memory claimed) {
+        Period storage period = periods[_periodId];
+
+        claimed = new bool[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            claimed[i] = _hasClaimedJurorFees(period, _juror, _tokens[i]);
+        }
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _token Address of the token to be queried
+    * @return Token amount corresponding to the governor
+    */
+    function getGovernorFees(uint256 _periodId, address _token) external view returns (uint256) {
+        Period storage period = periods[_periodId];
+        return period.governorFees[_token];
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _tokens List of token addresses to be queried
+    * @return List of token amounts corresponding to the governor
+    */
+    function getManyGovernorFees(uint256 _periodId, address[] calldata _tokens) external view returns (uint256[] memory amounts) {
+        Period storage period = periods[_periodId];
+
+        amounts = new uint256[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            amounts[i] = _getGovernorFees(period, _tokens[i]);
+        }
+    }
+
+    /**
+    * @dev Internal function to claim juror fees for a certain period
+    * @param _period Period being claimed
+    * @param _periodId Identification number of the period claiming fees for
+    * @param _juror Address of the juror claiming the fees
+    * @param _token Address of the token being claimed
+    * @param _amount Amount of tokens to be transferred to the juror
+    */
+    function _claimJurorFees(Period storage _period, uint256 _periodId, address payable _juror, address _token, uint256 _amount) internal {
+        if (_amount > 0) {
+            _period.claimedJurorFees[_juror][_token] = true;
```

Comment:
I'm not against this, as it saves gas, but we might consider moving this out of the `if` (or flip the `hasJurorClaimed()` as suggested above).

Otherwise, if you have nothing to claim, you could keep calling (or getting fooled into calling) functions relying on this check haha.

Reply:
I actually did both, moved this out of the if and change `hasGuardianClaimed` for `canGuardianClaim`. Addressed in 55bcbf4056e85ce958a64136d44ad7ea0038c304